add option in frodoProtocols to include a pulse to measure input resistance at the beginning of the trial

thulan-lily opened the pull request:
add option to have a pulse at the beginning of the trial in frodoProtocols
- after choosing stimulus, ask use whether they want to include a pulse at the beginning of the trial
- if yes, run new function currentStepPulse with hard-coded amplitude + duration values and append to stimulus command
- if no, runs stimulus without pulse

thulan-lily committed:
update frodoProtocols

frodoProtocols -update currentSteps stimulus function to be used in current sweep experiments

thulan-lily committed:
option in frodoProtocols to include a pulse at the beginning of the trial

add option to have a pulse at the beginning of the trial in frodoProtocols
- after choosing stimulus, ask use whether they want to include a pulse at the beginning of the trial
- if yes, run new function currentStepPulse with hard-coded amplitude + duration values and append to stimulus command
- if no, runs stimulus without pulse

evettita commented on frodoProtocols.m:
Great choice of where to do this edit, since this makes sure that all the DAQ handling that occurs in aquireTrial will just run correctly based on the length of stimulus.output!

evettita commented on frodoProtocols.m:
```diff
@@ -406,6 +417,23 @@
+
+BREAK_DURATION = 0.05; %sec
+
+preStepCommand = zeros(1, BREAK_DURATION * rigSettings.sampRate );
```
Could rename this to preAndPostStepCommand  or spacerStepCommand or something, NBD

evettita commented on frodoProtocols.m:
```diff
@@ -406,6 +417,23 @@
+
+pulseCommand = [preStepCommand stepCommand preStepCommand];
+
+pulseCommandArray = pulseCommand * rigSettings.command.currentClampExternalCommandGain; % send full command out, in Voltage for the daq to send
```
Looks prefect!   Unrelated to this commit, but is this gain value from the 700B now saved from the telegraph so we can confirm later if this was ever wrong?

thulan-lily committed:
rename preStepCommand in frodoProtocols currentStepPulse

rename preStepCommand to breakStepCommand in frodoProtocols currentStepPulse function
